src/platform/user/profile/vet360/components/AddressField/AddressForm.jsx

```diff
@@ -27,6 +42,30 @@ class AddressForm extends React.Component {
     focusElement('h5');
   }
 
+  onChange = async () => {
+    const { onInput } = this.props;
+    const {
+      stateCode,
+      city,
+      countryName,
+      zipCode,
+    } = this.state.addressFromRedux;
+    if (this.state.isMilitaryAddress) {
+      // If military base is unchecked, we don't want the user to lose their previously entered data
+      await onInput('countryName', countryName);
```

Comment:
This works. It would be cleaner to dispatch an `UPDATE_PROFILE_FORM_FIELD` action with the entire address here instead of updating each address form field individually in sequence, but I can live with this.